raiden-ts/src/channels/utils.ts

```diff
@@ -59,3 +65,31 @@ export function channelAmounts(channel: Channel) {
     partnerCapacity,
   };
 }
+
+/**
+ * Custom operator to wait & assert transaction success
+ *
+ * @param method - method name to use in logs
+ * @param error - ErrorCode to throw if transaction fails
+ * @param deps - object containing logger
+ * @returns operator function to wait for transaction and output hash
+ */
+export function assertTx(
+  method: string,
+  error: ErrorCodes,
+  { log }: Pick<RaidenEpicDeps, 'log'>,
+): OperatorFunction<ContractTransaction, Hash> {
+  return tx =>
```

Comment:
I don't get it, where does `tx` come from, some generic voodoo from `OperatorFunction`?

Reply:
@nephix It justs returns a function that takes a transaction as a parameter. Then when you put it in the `pipe()` pipe passes the value to the function. All these operator functions are for use in pipes. Does this help?

Reply:
Yes thanks a lot 🙏